### PR TITLE
[Scanner] Add concrete execution check for current execution

### PR DESCRIPTION
**Pull request by mkolodezny**
**What changed?**
Add concrete execution exists check for current execution

**Why?**
According to recent data we have in DLQ the most corrupted current executions do not have the corresponding concrete one.
The check will be run against DLQ messages and used by the scanner.  

**How did you test it?**
unit tests
Integration tests against Cassandra

### CODE REVIEW

File: common/persistence/cassandra/cassandraPersistence.go [hunk 1]
```diff
@@ -2053,6 +2063,38 @@ func (d *cassandraPersistence) GetCurrentExecution(request *p.GetCurrentExecutio
...
+	result := make(map[string]interface{})
+	if err := query.MapScan(result); err != nil {
+		if err == gocql.ErrNotFound {
+			return nil, &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("Workflow execution not found.  WorkflowId: %v",
+					request.WorkflowID),
+			}
+		} else if isThrottlingError(err) {
+			return nil, &workflow.ServiceBusyError{
+				Message: fmt.Sprintf("GetCurrentExecution operation failed. Error: %v", err),
```
Comment (yux0) [inline common/persistence/cassandra/cassandraPersistence.go:2087]: GetConcreteExecution

File: common/persistence/cassandra/cassandraPersistence.go [hunk 2]
```diff
@@ -2053,6 +2063,38 @@ func (d *cassandraPersistence) GetCurrentExecution(request *p.GetCurrentExecutio
...
+			}
+		}
+
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetCurrentExecution operation failed. Error: %v", err),
```
Comment (yux0) [inline common/persistence/cassandra/cassandraPersistence.go:2092]: GetConcreteExecution

File: common/persistence/cassandra/cassandraPersistence.go [hunk 3]
```diff
@@ -2053,6 +2063,38 @@ func (d *cassandraPersistence) GetCurrentExecution(request *p.GetCurrentExecutio
...
+		request.RunID,
+		defaultVisibilityTimestamp,
+		rowTypeExecutionTaskID)
+
+	result := make(map[string]interface{})
+	if err := query.MapScan(result); err != nil {
+		if err == gocql.ErrNotFound {
+			return nil, &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("Workflow execution not found.  WorkflowId: %v",
+					request.WorkflowID),
```
Comment (yux0) [inline common/persistence/cassandra/cassandraPersistence.go:2083]: may also include run id in the message

File: common/persistence/dataInterfaces.go [hunk 4]
```diff
@@ -1521,6 +1532,7 @@ type (
 		DeleteWorkflowExecution(request *DeleteWorkflowExecutionRequest) error
 		DeleteCurrentWorkflowExecution(request *DeleteCurrentWorkflowExecutionRequest) error
 		GetCurrentExecution(request *GetCurrentExecutionRequest) (*GetCurrentExecutionResponse, error)
+		GetConcreteExecution(request *GetConcreteExecutionRequest) (*GetConcreteExecutionResponse, error)
```
Comment (yux0) [inline common/persistence/dataInterfaces.go:1535]: Could we reuse GetWorkflowExecution()?

File: common/reconciliation/common/persistenceRetryer.go [hunk 5]
```diff
@@ -102,6 +102,23 @@ func (pr *persistenceRetryer) GetCurrentExecution(
 	return resp, nil
 }
 
+// GetCurrentExecution retries GetCurrentExecution
```
Comment (yux0) [inline common/reconciliation/common/persistenceRetryer.go:105]: GetConcreteExecution

File: common/reconciliation/invariants/concreteExecutionExists.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+func (c *concreteExecutionExist) Fix(execution interface{}) common.FixResult {
+	fixResult, checkResult := checkBeforeFix(c, execution)
+	if fixResult != nil {
+		return *fixResult
+	}
+	currentExecution, _ := execution.(*common.CurrentExecution)
+	if err := c.pr.DeleteCurrentWorkflowExecution(&persistence.DeleteCurrentWorkflowExecutionRequest{
+		DomainID:   currentExecution.DomainID,
+		WorkflowID: currentExecution.WorkflowID,
+		RunID:      currentExecution.RunID,
```
Comment (yux0) [inline common/reconciliation/invariants/concreteExecutionExists.go:100]: current run id

**Comment by yux0** [comment]
If we decide to add this new query GetConcreteExecution, it may be worth to add some db integration tests

File: common/reconciliation/invariants/concreteExecutionExists.go [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+// NewConcreteExecutionExists returns a new invariant for checking concrete execution
+func NewConcreteExecutionExists(
+	pr common.PersistenceRetryer,
+) common.Invariant {
+	return &concreteExecutionExist{
+		pr: pr,
+	}
+}
+
+func (c *concreteExecutionExist) Check(execution interface{}) common.CheckResult {
```
Comment (yux0) [inline common/reconciliation/invariants/concreteExecutionExists.go:46]: The logic looks good. Maybe worth to add a unit test

File: common/persistence/dataInterfaces.go (hunk 4 above)
Comment (mkolodezny) [reply common/persistence/dataInterfaces.go:1535]: We could, but it selects and does lots of unnecessary stuff while we need to keep lookups as lite as possible for the scanner.

File: common/persistence/cassandra/cassandraPersistence.go [hunk 8]
```diff
@@ -440,6 +440,16 @@ workflow_state = ? ` +
 		`and visibility_ts = ? ` +
 		`and task_id = ?`
 
+	templateIsWorkflowExecutionExistsQuery = `SELECT shard_id, type, domain_id, workflow_id, run_id, visibility_ts, task_id ` +
```
Comment (yux0) [inline common/persistence/cassandra/cassandraPersistence.go:443]: The query data may not useful for checking the workflow existence?

**Pull request merged**